src/org/labkey/test/tests/ParsingPatternForDateTest.java

```diff
@@ -65,54 +78,121 @@ private void doSetup() throws IOException, CommandException
+        String dateTimePattern = "ddMMMyyyy:HH:mm:ss";
+        String datePattern = "mm/dd/yy";
+        String timePattern = "hh:mm a";
+        setAdditionalParsingPatterns(dateTimePattern, datePattern, timePattern);
+
+        List<Map<String, String>> expectedTableValues = new ArrayList<>();
 
-        log("Update a row with a date in a non-standard format");
+        log("Update a row with values in the additional parsing format.");
         goToProjectHome();
         clickAndWait(Locator.linkWithText(dateList));
         DataRegionTable listTable = new DataRegionTable("query", getDriver());
         listTable.clickEditRow(0);
-        setFormElement(Locator.name("quf_dateCol"), "25Nov2020:24:23:00");
+
+        Map<String, String> setValuesMap = new HashMap<>();
+        setValuesMap.put("dateTimeCol", "2020-11-26 00:23");
+        setValuesMap.put("dateCol", "2020-11-25");
+        setValuesMap.put("timeCol", "00:23:00");
+
+        for(var entry : setValuesMap.entrySet())
+        {
+            setFormElement(Locator.name(String.format("quf_%s", entry.getKey())), entry.getValue());
+        }
         clickButton("Submit");
 
-        checker().verifyEquals("Incorrect date parsed while editing the row", "2020-11-26 00:23", listTable.getDataAsText(0, "dateCol"));
+        // Add the name to the values expected in the grid.
+        setValuesMap.put("name", "First");
+
+        expectedTableValues.add(setValuesMap);
+
+        log("Insert a row with a data in the additional parsing format.");
 
-        log("Insert a row with a date in a non-standard format");
         listTable.clickInsertNewRow();
-        setFormElement(Locator.name("quf_name"), "Second");
-        setFormElement(Locator.name("quf_dateCol"), "23Nov2020:12:23:34");
-        clickButton("Submit");
 
-        listTable.setFilter("name", "Equals", "Second");
-        checker().verifyEquals("Incorrect date parsed while inserting row", "2020-11-23 12:23", listTable.getDataAsText(0, "dateCol"));
+        setValuesMap = Map.of(
+                "name", "Second",
+                "dateTimeCol", "2020-11-23 12:23",
+                "dateCol", "2020-11-23",
+                "timeCol", "12:23:00"
+        );
+
+        expectedTableValues.add(setValuesMap);
 
-        log("Bulk import with some dates in non-standard format");
+        for(var entry : setValuesMap.entrySet())
+        {
+            setFormElement(Locator.name(String.format("quf_%s", entry.getKey())), entry.getValue());
+        }
+        clickButton("Submit");
+
+        log("Bulk import with values in additional parsing patterns.");
         listTable.clearAllFilters();
         listTable.clickImportBulkData();
-        click(Locator.tagWithText("h3", "Upload file (.xlsx, .xls, .csv, .txt)"));
+        click(Locator.tagContainingText("h3", "Upload file"));
+
+        // Note: the date and time columns in this xlsx are formatted as date and time types. Excel does not have a date-time format, that is a string.
```

Comment:
Hmm.. Date/Time/Datetime in excel are all "Numbers", with different format. The field should be a "Number" type, not a string type.

Reply:
Though it would be good to have separate coverage that test import from excel with "String" type date/time/datetime too.